docs/stubs.rst

```diff
@@ -0,0 +1,1069 @@
+
+Instance attributes and class variables follow the same recommendations as
+module level attributes:
+
+Yes::
+
+    class Foo:
+        c: ClassVar[str]
+        x: int
+
+No::
+
+    class Foo:
+        c: ClassVar[str] = ""
+        d: ClassVar[int] = ...
+        x = 4
+        y: int = ...
+
+Functions and Methods
+---------------------
+
+Use the same argument names as in the implementation, because
+otherwise using keyword arguments will fail. Of course, this
+does not apply to positional-only arguments, which are marked with a double
+underscore.
+
+Use the ellipsis literal ``...`` in place of actual default argument
+values. Use an explicit ``Optional`` annotation instead of
+a ``None`` default.
+
+Yes::
+
+    def foo(x: int = ...) -> None: ...
+    def bar(y: Optional[str] = ...) -> None: ...
+
+No::
+
+    def foo(x: int = 0) -> None: ...
+    def bar(y: str = None) -> None: ...
+    def baz(z: Optional[str] = None) -> None: ...
+
+Do not annotate ``self`` and ``cls`` in method definitions, except when
+referencing a type variable.
+
+Yes::
+
+    _T = TypeVar("_T")
+    class Foo:
+        def bar(self) -> None: ...
+        @classmethod
+        def create(cls: type[_T]) -> _T: ...
+
+No::
+
+    class Foo:
+        def bar(self: Foo) -> None: ...
+        @classmethod
+        def baz(cls: type[Foo]) -> int: ...
+
+The bodies of functions and methods should consist of only the ellipsis
+literal ``...`` on the same line as the closing parenthesis and colon.
+
+Yes::
+
+    def to_int1(x: str) -> int: ...
+    def to_int2(
+        x: str,
+    ) -> int: ...
+
+No::
+
+    def to_int1(x: str) -> int:
+        return int(x)
+    def to_int2(x: str) -> int:
+        ...
+    def to_int3(x: str) -> int: pass
+
+Private Definitions
+-------------------
+
+Type variables, type aliases, and other definitions that should not
+be used outside the stub should be marked as private by prefixing them
+with an underscore.
+
+Yes::
+
+    _T = TypeVar("_T")
+    _DictList = dict[str, list[Optional[int]]]
+
+No::
+
+    T = TypeVar("T")
+    DictList = dict[str, list[Optional[int]]]
+
+Language Features
+-----------------
+
+Use the latest language features available as outlined
+in the Syntax_ section, even for stubs targeting older Python versions.
+Do not use quotes around forward references and do not use ``__future__``
+imports.
+
+Yes::
+
+    class Py35Class:
+        x: int
+        forward_reference: OtherClass
+    class OtherClass: ...
+
+No::
+
+    class Py35Class:
+        x = 0  # type: int
+        forward_reference: 'OtherClass'
+    class OtherClass: ...
+
+Types
+-----
+
+Generally, use ``Any`` when a type cannot be expressed appropriately
+with the current type system or using the correct type is unergonomic.
+
+Use ``float`` instead of ``Union[int, float]``.
+Use ``None`` instead of ``Literal[None]``.
+For argument types,
+use ``bytes`` instead of ``Union[bytes, memoryview, bytearray]``.
```

Comment:
I think this needs a bit more explaining, or a link to somewhere where it's documented, since it does not match runtime behaviour:

```
>>> isinstance(bytearray(), bytes)
False
```

Reply:
This should probably be more prominent in the rest of the typing documentation (it is explained in the typing module docs), but the style guide is not really the correct place to explain it, just as `float == float | int` is not explained.